upb/def.c

```diff
@@ -364,19 +387,24 @@ bool upb_def_freeze(upb_def *const* defs, int n, upb_status *s) {
    * limit. */
   maxdepth = UPB_MAX_MESSAGE_DEPTH * 2;
 
-  /* Validation all passed; freeze the defs. */
-  ret = upb_refcounted_freeze((upb_refcounted * const *)defs, n, s, maxdepth);
```

Comment:
was this cast unsafe?  Otherwise it seems nice to have the type safety that the objs are upb_defs